test/testdata/generate.sh

```diff
@@ -54,7 +54,11 @@ function scaffold_test_project {
       $kb create webhook --group crew --version v1 --kind Captain --defaulting --programmatic-validation --force
     fi
 
-    $kb create api --group crew --version v1 --kind FirstMate --controller=true --resource=true --make=false
+    if [ $project == "project-v2" ]; then
+      $kb create api --plugins="go/v2,declarative" --group crew --version v1 --kind FirstMate --controller=true --resource=true --make=false
+    else
+      $kb create api --plugins="go/v3,declarative" --group crew --version v1 --kind FirstMate --controller=true --resource=true --make=false
+    fi
```

Comment:
```suggestion
    # todo: improve the plugins api to address #1941 
    # if [ $project == "project-v2" ]; then
    #  $kb create api --plugins="go/v2,declarative" --group crew --version v1 --kind FirstMate --controller=true --resource=true --make=false
    # else
    #  $kb create api --plugins="go/v3,declarative" --group crew --version v1 --kind FirstMate --controller=true --resource=true --make=false
    # fi
```

To close #1941 it is missing: 
- By informing --plugins to the subcommand it is missing track the info in the PROJECT file.
- Add a help text for declarative plugin to allow us to check how the help text will be with `$ kubebuilder create api --plugins="go/v3,declarative"`
- shows the --plugins option for all subcommands with the cli help
- Improve the plugin subcommand overwrite behaviour to work as we spoke out in the issue #1941 project layout + values informed on the flag. See the motivations clarifies in the issue.

Reply:
> * By informing --plugins to the subcommand it is missing track the info in the PROJECT file.

`declarative` is not tracking any info as you mention, will add a list of resources that were scaffolded with this plugin.

> * Add a help text for declarative plugin to allow us to check how the help text will be with `$ kubebuilder create api --plugins="go/v3,declarative"`

Will do

> * shows the --plugins option for all subcommands with the cli help

This is already working:
```
$ kubebuilder create api --plugins go/v3,declarative --help
[...]
Global Flags:
      --plugins strings   plugin keys to be used for this subcommand execution
```

Reply:
Done